testnet/stacks-node/src/nakamoto_node/stackerdb_listener.rs

```diff
@@ -0,0 +1,444 @@
+                    return Err(NakamotoNodeError::SigningCoordinatorFailure(
+                        "StackerDB event receiver disconnected".into(),
+                    ));
+                }
+            };
+
+            // check to see if this event we got is a signer event
+            let is_signer_event =
+                event.contract_id.name.starts_with(SIGNERS_NAME) && event.contract_id.is_boot();
+
+            if !is_signer_event {
+                debug!("StackerDBListener: Ignoring StackerDB event for non-signer contract"; "contract" => %event.contract_id);
+                continue;
+            }
+
+            let modified_slots = &event.modified_slots.clone();
+
+            let Ok(signer_event) = SignerEvent::<SignerMessageV0>::try_from(event).map_err(|e| {
+                warn!("StackerDBListener: Failure parsing StackerDB event into signer event. Ignoring message."; "err" => ?e);
+            }) else {
+                continue;
+            };
+            let SignerEvent::SignerMessages(signer_set, messages) = signer_event else {
+                debug!("StackerDBListener: Received signer event other than a signer message. Ignoring.");
+                continue;
+            };
+            if signer_set != self.signer_set {
+                debug!(
+                    "StackerDBListener: Received signer event for other reward cycle. Ignoring."
+                );
+                continue;
+            };
+            let slot_ids = modified_slots
+                .iter()
+                .map(|chunk| chunk.slot_id)
+                .collect::<Vec<_>>();
+
+            debug!("StackerDBListener: Received messages from signers";
+                "count" => messages.len(),
+                "slot_ids" => ?slot_ids,
+            );
+
+            for (message, slot_id) in messages.into_iter().zip(slot_ids) {
+                let Some(signer_entry) = &self.signer_entries.get(&slot_id) else {
+                    return Err(NakamotoNodeError::SignerSignatureError(
+                        "Signer entry not found".into(),
+                    ));
+                };
+                let Ok(signer_pubkey) = StacksPublicKey::from_slice(&signer_entry.signing_key)
+                else {
+                    return Err(NakamotoNodeError::SignerSignatureError(
+                        "Failed to parse signer public key".into(),
+                    ));
+                };
+
+                match message {
+                    SignerMessageV0::BlockResponse(BlockResponse::Accepted(accepted)) => {
+                        let BlockAccepted {
+                            signer_signature_hash: block_sighash,
+                            signature,
+                            metadata,
+                            response_data,
+                        } = accepted;
+                        let tenure_extend_timestamp = response_data.tenure_extend_timestamp;
+
+                        let (lock, cvar) = &*self.blocks;
```

Comment:
This is something a `Comms` struct could hide from the caller.

Reply:
Improved in `StackerDBListenerComms`.